easybuild/easyblocks/t/tensorflow.py

```diff
@@ -0,0 +1,260 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing TensorFlow, implemented as an easyblock
+"""
+import glob
+import os
+import stat
+import tempfile
+
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, apply_regex_substitutions, mkdir, resolve_path
+from easybuild.tools.filetools import which, write_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+
+INTEL_COMPILER_WRAPPER = """#!/bin/bash
+export INTEL_LICENSE_FILE='%(intel_license_file)s'
+%(compiler_path)s "$@"
+"""
+
+
+class EB_TensorFlow(PythonPackage):
+    """Support for building/installing TensorFlow."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            # see https://developer.nvidia.com/cuda-gpus
+            'cuda_compute_capabilities': [[], "List of CUDA compute capabilities to build with", CUSTOM],
+            'with_mkl_dnn': [True, "Make TensorFlow use Intel MKL-DNN", CUSTOM],
+        }
+        return PythonPackage.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configuration procedure for TensorFlow."""
+
+        tmpdir = tempfile.mkdtemp(suffix='-bazel-configure')
+
+        # create wrapper for icc to make sure location of license server is available...
+        # cfr. https://github.com/bazelbuild/bazel/issues/663
+        if self.toolchain.comp_family() == toolchain.INTELCOMP:
+            icc_wrapper_txt = INTEL_COMPILER_WRAPPER % {
+                'compiler_path': which('icc'),
+                'intel_license_file': os.getenv('INTEL_LICENSE_FILE', os.getenv('LM_LICENSE_FILE')),
+            }
+            icc_wrapper = os.path.join(tmpdir, 'bin', 'icc')
+            write_file(icc_wrapper, icc_wrapper_txt)
+            adjust_permissions(icc_wrapper, stat.S_IXUSR)
+            env.setvar('PATH', ':'.join([os.path.dirname(icc_wrapper), os.getenv('PATH')]))
+            self.log.info("Using wrapper script for 'icc': %s", which('icc'))
+
+        self.prepare_python()
+
+        cuda_root = get_software_root('CUDA')
+        cudnn_root = get_software_root('cuDNN')
+        jemalloc_root = get_software_root('jemalloc')
+        opencl_root = get_software_root('OpenCL')
```

Comment:
There isn't any `OpenCL` easyconfig as far as I can see. Besides, the `OpenCL` runtime for intel processors needs root permissions, so it won't be made available any time soon through EB.

Reply:
Well, ok, but I'll leave this in anyway, to avoid hardcoding building without `OpenCL` support.

At least this way there is an option to built TF with OpenCL support (for example via http://easybuild.readthedocs.io/en/latest/Using_external_modules.html).